docs/misconfiguration/policy/custom.md

```diff
@@ -0,0 +1,180 @@
+| ------------------ | ------------------------------------| :-----------: | :----------------: |:---------------: |
+| id                 | Any characters                      | N/A           | :material-check:   | :material-check: |
+| title              | Any characters                      | N/A           | :material-check:   | :material-check: |
+| severity           | `LOW`, `MEDIUM`, `HIGH`, `CRITICAL` | UNKNOWN       | :material-check:   | :material-check: |
+| type               | Any characters                      | N/A           | :material-check:   | :material-check: |
+| description        | Any characters                      |               | :material-close:   | :material-check: |
+| recommended_actions| Any characters                      |               | :material-close:   | :material-check: | 
+| url                | Any characters                      |               | :material-close:   | :material-check: |
+
+Some fields are displayed in scan results.
+
+``` bash
+deployment.yaml (kubernetes)
+============================
+Tests: 28 (SUCCESSES: 14, FAILURES: 14, EXCEPTIONS: 0)
+Failures: 14 (HIGH: 1)
+
++---------------------------+------------+-------------------------------------+----------+------------------------------------------+
+|           TYPE            | MISCONF ID |                CHECK                | SEVERITY |                 MESSAGE                  |
++---------------------------+------------+-------------------------------------+----------+------------------------------------------+
+| Custom Kubernetes Check   |   ID001    | Deployment not allowed              |   LOW    | Found deployment 'test' but deployments  |
+|                           |            |                                     |          | are not allowed                          |
++---------------------------+------------+-------------------------------------+----------+------------------------------------------+
+```
+
+### Input
+You can specify input format via `__rego_input__`.
+All fields under `__rego_input` are optional.
+
+!!!example
+    ``` rego
+    __rego_input__ := {
+        "combine": false,
+        "selector": [
+            {"type": "kubernetes"},
+        ],
+    }
+    ```
+
+`combine` (boolean)
+: The details is [here](../advanced/combine.md)  
+
+`selector` (array)
+:   This option filters the input by config types. 
+    In the above example, Trivy passes only Kubernetes files to this policy.
+    Even if Dockerfile exists in the specified directory, it will not be passed to the policy as input.
+
+    `type` accepts `kubernetes`, `dockerfile`, `ansible`, `cloudformation`, `json`, `yaml`, toml`, or `hcl`.
+
+    !!! note
+        If the configuration type is identified, the type will be overwritten, even if the file is genera format such as JSON or YAML.  
+        For example, `deployment.yaml` will be handled as `kubernetes`, not `yaml`.
```

Comment:
maybe instead of this "note", add to the selector description above, that it's not just selecting which files to pass to the policy, but also affects how they will be parsed

Reply:
`--file-pattern` does affect how files are parsed, but `selector` doesn't affect.